Bindings/Java/OpenSimJNI/Test/testContext.cpp

```diff
@@ -222,6 +222,19 @@ int main()
     std::string pathAfterTypeChangeToViaInXML = dTRIlong->updGeometryPath().dump();
     std::cout << pathAfterTypeChangeToViaInXML << endl;
  
+    // edit sockets
+    context->cacheModelAndState();
+    Joint& shoulder = model->updComponent<Joint>("r_shoulder");
+    AbstractSocket& socket = shoulder.updSocket("child_frame");
+    socket.setConnecteeName("ground"); // This create a loop and throws excption at initSystem
+    try {
+        context->restoreStateFromCachedModel();
+    }
+    catch (...) {
+        // undo the change
```

Comment:
Not something like this to undo the change?
```cpp
const std::string originalConnecteeName = socket.getConnecteeName();
socket.setConnecteeName("ground");
...
socket.setConnecteeName(originalConnecteeName);
```

Reply:
:+1: Much nicer! I'll update.